### PR TITLE
prevent rootfs writing from corrupt backup GPT partition

### PR DESCRIPTION
**Pull request by sprhawk**
Old code will destroy backup partition table of GPT. And cause GPT tools or linux boot up complains error state in GPT partition

This will preserve it

### CODE REVIEW

**Comment by wzyy2** [comment]
Thanks, It looks great.
will it expanded properly to fill all available space after linux boot up?

File: classes/rockchip-gpt-img.bbclass [hunk 1]
```diff
@@ -83,8 +83,17 @@ IMAGE_CMD_rockchip-gpt-img () {
...
+	# but this will overrite the backup table of GPT
+	# will cause corruption error for GPT
+	IMG_ROOTFS_SIZE=$(stat -L --format="%s" ${IMG_ROOTFS})
+
+	GPTIMG_MIN_SIZE=$(expr $IMG_ROOTFS_SIZE + \( ${LOADER1_SIZE} + ${RESERVED1_SIZE} + ${RESERVED2_SIZE} + ${LOADER2_SIZE} + ${ATF_SIZE} + ${BOOT_SIZE} + 35 \) \* 512 )
+
+	GPT_IMAGE_SIZE=$(expr $GPTIMG_MIN_SIZE \/ 1024 \/ 1024 + 1)
+
 	# Initialize sdcard image file
-	dd if=/dev/zero of=${GPTIMG} bs=1M count=0 seek=${GPTIMG_SIZE}
```
Comment (wzyy2) [inline classes/rockchip-gpt-img.bbclass]: It should keep using "count=0 seek=$XXXXX", it's faster than using "count=XXXX".

File: classes/rockchip-gpt-img.bbclass [hunk 2]
```diff
@@ -109,7 +118,7 @@ create_rk_image () {
 	parted -s ${GPTIMG} set 6 boot on
 
 	# Create rootfs partition
-	parted -s ${GPTIMG} unit s mkpart rootfs ${ROOTFS_START} 100%
```
Comment (wzyy2) [inline classes/rockchip-gpt-img.bbclass]: What does this change mean?

**Comment by sprhawk** [comment]
I have changed the dd command back

**Comment by wzyy2** [comment]
Thanks.

**Pull request merged**